fastparquet/test/test_api.py

```diff
@@ -193,10 +207,10 @@ def test_request_nonexistent_column(tempdir):
 def test_read_multiple_no_metadata(tempdir):
     df = pd.DataFrame({'x': [1, 5, 2, 5]})
     write(tempdir, df, file_scheme='hive', row_group_offsets=[0, 2])
-    os.unlink(os.path.join(tempdir, '_metadata'))
-    os.unlink(os.path.join(tempdir, '_common_metadata'))
+    os.unlink(join_path(tempdir, '_metadata'))
+    os.unlink(join_path(tempdir, '_common_metadata'))
     import glob
-    flist = list(sorted(glob.glob(os.path.join(tempdir, '*'))))
+    flist = list(sorted(map(join_path, glob.glob(join_path(tempdir, '*')))))
```

Comment:
The API needs to be able to work when supplied with windows paths too (this test and below). Probably the test for a windows path only needs to happen if running on windows (r'\' is rarely used in posix paths but it is legal).

Reply:
Yes, this works on windows paths;  `glob.glob` has no problem accepting Linux paths on a windows machine, so `join_path(tempdir, '*')` is okay.  `glob.glob` returns windows paths, and they must be converted back to standard form, hence `map(join_path, ...`

Reply:
Sorry, you misunderstand. The point it, the user may give windows paths to the ParquetFile constructor, so somewhere we must test something that looks like
```
ParquetFile([r'C:\path\file1', r'C:\path\file2])
```
(whether those paths are generated with glob or not is immaterial).

Reply:
Okay, I *think* I got you now. My next push will revert all the tests (except the ones I added).  These tests use native filenames, including the `ParquetFile([r'C:\path\file1', r'C:\path\file2])` you are concerned about.  All tests (except the two I added as issues) pass on windows

Reply:
I don't know if you did it already, but there's no need ot revert all tests, so long as windows paths are also tested elsewhere.

Reply:
It is done already, and it is a good thing.  By keeping the tests the same, we are more confident that these test do not break something.  Plus, the original tests used Window `os.sep` on windows, meaning the diversity of testing is greater.